x-pack/test/alerting_api_integration/security_and_spaces/group2/tests/alerting/rbac_legacy.ts

```diff
@@ -103,32 +102,42 @@ export default function alertTests({ getService }: FtrProviderContext) {
               // these cases were invalid pre 7.10.0 and remain invalid post 7.10.0
               break;
             case 'space_1_all at space1':
-            case 'superuser at space1':
             case 'space_1_all_with_restricted_fixture at space1':
+            case 'superuser at space1':
               await resetTaskStatus(migratedAlertId);
               await ensureLegacyAlertHasBeenMigrated(migratedAlertId);
 
               await updateMigratedAlertToUseApiKeyOfCurrentUser(migratedAlertId);
+              await rescheduleTask(migratedAlertId);
 
-              await ensureAlertIsRunning();
-
-              await updateAlertSoThatItIsNoLongerLegacy(migratedAlertId);
+              await alertUtils.disable(migratedAlertId);
+              await retry.try(async () => {
+                const updateResponse = await updateAlertSoThatItIsNoLongerLegacy(migratedAlertId);
+                expect(updateResponse.statusCode).to.eql(200);
+              });
 
               // update alert as user with privileges - so it is no longer a legacy alert
-              const updatedKeyResponse = await alertUtils.getUpdateApiKeyRequest(migratedAlertId);
-              expect(updatedKeyResponse.statusCode).to.eql(204);
-
+              await retry.try(async () => {
```

Comment:
nit: maybe we could move the `retry.try` into the `alertUtils.getUpdateApiKeyRequest` helper function?

Reply:
i'd rather skipping this.
This function is used 2 more times without retry.